### ISSUE TITLE
Parse city names even when there's no comma after the address line

### ISSUE THREAD

**Issue by markstos** [thrustlabs/contact-parser#15]
I've got some real-world data I'd like to parse, which looks like this:

```
12892 East Center Avenue Aurora, CO  80012
```

The very-useful comma between the address line and the city name is missing. 

[ESRI's geocoder can parse this](http://geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer/find?f=pjson&text=12892%20East%20Center%20Avenue%20Aurora,%20CO%2080012) but `contact-parser` can't currently. 

Here's how I think it could be fixed:

 1. After the current parsing pass, check if the city name is blank and the street type in the address line not recognized.
 2. In that case, walk backwards in the address line looking for a recognized street type (like "Street", "St',  "Road", etc). If one is found assume the characters after the street type are the city name, and move them to the empty city name field. 

There are other open source libraries that we can borrow the list of US street types from.

I'm open to contributing a PR for this, but wanted to discuss it first. 

As an aside, has there been consideration to porting the code from CoffeeScript to ES6? ES6 offers a number of features of CoffeeScript, and there are also transpilers to convert it to ES5. I think it could increase contributions, since ES6 will be more common in the future than CoffeeScript. 

The LogEntries client is an example of a module written in ES6: https://github.com/logentries/le_node

**Comment by markstos** [comment]
OK, I noticed that all addresses without the comma fail. For example, this one is parsed correctly:

```
  '1234 South Clayton Street Denver, CO 80210'
```

Looks like the difference comes down to the definition of `streetNameRegex`:

https://github.com/thrustlabs/contact-parser/blob/master/src/contact-parser.coffee#L14

... which understands "Street", but not "Avenue". 

That means a fix is potentially easy. We just have update `streetNameRegex` to be smarter, based on a list of possible street names from another project. I'm on it!

**Comment by jasondoucette** [comment]
Thanks Mark, and apologies for the delay - I've been away from js for a bit so my dev environment is a little mangled, hence the comments on the PR rather than just making the changes. I should be able to resolve this tomorrow if you're busy.

As for ES6, it's not on my radar atm based on my current work environment - I'm not married to CoffeeScript but don't have the resources to do a port right now.

**Comment by jasondoucette** [comment]
PR merged, thanks again - I'll package it up as 0.0.6
